tests/test_user_task_registration.py

```diff
@@ -106,3 +106,44 @@ class MyCOPA(COPA):
         monkeypatch.syspath_prepend(tmpdir)
         load_extra_tasks([task_file])
         assert TaskName.from_name("MyCOPA")
+
+
+def test_user_task_registration_with_repeated_names(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test that loading user tasks with duplicate names raises an error."""
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Fyi: there is a `tmp_path` pytest fixture that you can use to avoid the boilder plate of creating tmp directories.